src/LegacyThumbnailMigrationHelper.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace SilverStripe\Assets;
+
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use SilverStripe\Assets\Dev\TestAssetStore;
+use SilverStripe\Assets\FilenameParsing\FileResolutionStrategy;
+use SilverStripe\Assets\Storage\AssetStore;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Environment;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DataQuery;
+use SilverStripe\Versioned\Versioned;
+
+/**
+ * Service to migrate legacy format thumbnails, to avoid regenerating them on demand.
+ * Related to SilverStripe\AssetAdmin\Helper\ImageThumbnailHelper,
+ * which proactively generates the (new) thumbnails required for asset-admin previews.
+ *
+ * Migrates thumbnails regardless whether their original file still exists,
+ * since they might still be hot-linked. Relies on the legacy file format redirections
+ * introduced in 4.3.3 and 4.4.0 for those hot-links to continue resolving.
+ *
+ * Example file format:
+ * Before (3.x): `assets/my-folder/_resampled/PadWzYwLDYwLCJGRkZGRkYiLDBd/FillWzYwLDMwXQ/my-image.jpg`
+ * After (4.x): `assets/my-folder/0ec62bd1c4/my-image__PadWzYwLDYwLCJGRkZGRkYiLDBd_CropHeightWzMwXQ.jpg`
+ *
+ * Limitations:
+ * - Does not migrate legacy thumbnails where the original file or folder
+ *   has been renamed since an earlier 4.x migration run
+ * - Does not filter out unused CMS thumbnails (they're using a new size now)
+ */
+class LegacyThumbnailMigrationHelper
```

Comment:
We're starting to have quite a few migration test in assets. I'm starting to think maybe we should move them all to their own sub-namespace and alias the existing ones.

Reply:
Done!